homeassistant/components/switch/bbb_gpio.py

```diff
@@ -0,0 +1,92 @@
+"""
+Allows to configure a switch using BBB GPIO.
+
+Switch example for two GPIOs pins P9_12 and P9_42
+Allowed GPIO pin name is GPIOxxx or Px_x
+
+switch:
+  - platform: bbb_gpio
+    ports:
+      GPIO0_7: LED Red
+      P9_12: LED Green
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA
+import homeassistant.components.bbb_gpio as bbb_gpio
+from homeassistant.const import DEVICE_DEFAULT_NAME
+from homeassistant.helpers.entity import ToggleEntity
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['bbb_gpio']
+
+CONF_PULL_MODE = 'pull_mode'
+CONF_PORTS = 'ports'
+CONF_INVERT_LOGIC = 'invert_logic'
+
+DEFAULT_INVERT_LOGIC = False
+
+_SWITCHES_SCHEMA = vol.Schema({
+    cv.string: cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PORTS): _SWITCHES_SCHEMA,
+    vol.Optional(CONF_INVERT_LOGIC, default=DEFAULT_INVERT_LOGIC): cv.boolean,
```

Comment:
I would expect invert logic to be an option that can be specified for each individual port.